iflist_test.go

```diff
@@ -0,0 +1,33 @@
+package nmap
+
+import (
+	"testing"
+)
+
+func TestScanner_GetInterfaceList(t *testing.T) {
+	scanner, err := NewScanner(WithBinaryPath("tests/scripts/fake_nmap_iflist.sh"))
+	if err != nil {
+		t.Error(err)
+	}
+
+	result, err := scanner.GetInterfaceList()
+	if err != nil {
+		t.Error(err)
+	}
+
+	if len(result.Interfaces) != 2 {
+		t.Error("There should be 2 interfaces inside")
+	}
+
+	if len(result.Routes) != 2 {
+		t.Error("There should be 2 routes inside")
+	}
+}
+
+func TestConvertInterface(t *testing.T) {
+	i := convertInterface("lo     (lo)     127.0.0.1/8                               loopback down 65536")
+
+	if i.Up {
+		t.Error("Interface should be down")
+	}
+}
```

Comment:
IIRC Cameradar uses the testify framework, so this can be simplified to

```suggestion
func TestScanner_GetInterfaceList(t *testing.T) {
	scanner, err := NewScanner(WithBinaryPath("tests/scripts/fake_nmap_iflist.sh"))
	require.NoError(t)

	result, err := scanner.GetInterfaceList()
	
	require.NoError(t)
	require.NotNil(t, result)
	assert.Len(t, result.Interfaces, 2)
	assert.Len(t, result.Routes, 2)
}

func TestConvertInterface(t *testing.T) {
	i := convertInterface("lo     (lo)     127.0.0.1/8                               loopback down 65536")

	assert.True(t, i.Up)
}
```

And additional checks should probably be done in the conversion function to cover each path of the conversion, not only whether it's up or not.

Reply:
I like the testify framework and will use it.